### DISCUSSION TITLE
User Sign In Taking A Long Time

### DISCUSSION THREAD

**Discussion by BenJarrett** [nss-evening-cohort-14/evening-client-side#313] [Capstones!] [answered]
Ben Jarrett

When I choose the user google email to sign in with, authentication takes a very long time.
After about 20 seconds, I get a pop up window stating "Page Unresponsive" You can for it become responsive or exit the page. Wait or Exit Page". 
If I click the "Wait" button, the user never logs in. If I hit the "Exit Page" button and reload the page, the user is signed in. 

There are no error messages that appear. 

I have tried waiting this out for a day, thinking that maybe it is something with firebase, but it is still happening.

https://www.npmjs.com/package/react-google-login
https://blog.logrocket.com/user-authentication-firebase-react-apps/
https://firebase.google.com/docs/auth/web/auth-state-persistence
https://www.freecodecamp.org/news/google-login-with-react-native-and-firebase/

https://github.com/BenJarrett/FrontEnd-Capstone/blob/adding-user-auth-v2/src/App/index.js

**Comment by BenJarrett** [comment, marked as answer]
I found the answer! 

My useEffect in my app index was improperly set up.
This was how it was setup.
It was causing an infinite loop when I passed user into my views. 

```
useEffect(() => {
    firebase.auth().onAuthStateChanged((authed) => {
      if (authed) {
        const userInfoObj = {
          fullName: authed.displayName,
          profileImage: authed.photoURL,
          uid: authed.uid,
          user: authed.email.split('@gmail.com')[0]
        };
        setUser(userInfoObj);
      } else if (user || user === null) {
        setUser(false);
      }
    });
  }, );
  ```
I needed to add an empty array before the last parenthesis!

``` 
useEffect(() => {
    firebase.auth().onAuthStateChanged((authed) => {
      if (authed) {
        const userInfoObj = {
          fullName: authed.displayName,
          profileImage: authed.photoURL,
          uid: authed.uid,
          user: authed.email.split('@gmail.com')[0]
        };
        setUser(userInfoObj);
      } else if (user || user === null) {
        setUser(false);
      }
    });
  }, []);
``` 
Passing an empty array as the second argument to useEffect makes it only run on mount and unmount, thus stopping any infinite loops

**Reply by drteresavasquez** [reply]
That dependency array will get you every time 😊
